soh/src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -7384,7 +7384,8 @@ s32 Player_ActionHandler_2(Player* this, PlayState* play) {
             }
 
             if ((this->heldActor == NULL) || Player_HoldsHookshot(this)) {
-                if ((interactedActor->id == ACTOR_BG_TOKI_SWD) && LINK_IS_ADULT) {
+                if ((interactedActor->id == ACTOR_BG_TOKI_SWD) && LINK_IS_ADULT &&
+                    !(Randomizer_GetSettingValue(RSK_SHUFFLE_MASTER_SWORD && !CHECK_OWNED_EQUIP(EQUIP_TYPE_SWORD, EQUIP_INV_SWORD_MASTER)))) {
```

Comment:
This should have a `IS_RANDO` before checking `Randomizer_GetSettingvalue` to account for vanilla
Also this function call looks mangled, its doing an AND with the RSK_ value which means its not returning the option value correctly, you probably missed a parenthesis after the RSK_ like you do below.

Reply:
Fixed